File: src/main/java/com/efedotov/vaultly/dto/user/UpdateKeysRequest.java
package com.efedotov.vaultly.dto.user;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class UpdateKeysRequest {
    @NotBlank
    private String publicKey;
    @NotBlank
    private String privateKeyEncrypted;
    @NotBlank
    private String currentPassword;
}

File: lib/server/model/user/update_keys_request.dart
class UpdateKeysRequest {
  final String publicKey;
  final String privateKeyEncrypted;
  final String currentPassword;

  UpdateKeysRequest({
    required this.publicKey,
    required this.privateKeyEncrypted,
    required this.currentPassword,
  });

  factory UpdateKeysRequest.fromJson(Map<String, dynamic> json) {
    return UpdateKeysRequest(
      publicKey: json['publicKey'] as String,
      privateKeyEncrypted: json['privateKeyEncrypted'] as String,
      currentPassword: json['currentPassword'] as String,
    );
  }

  Map<String, dynamic> toJson() => {
    'publicKey': publicKey,
    'privateKeyEncrypted': privateKeyEncrypted,
    'currentPassword': currentPassword,
  };
}
